tools/model_tools/src/openvino/model_zoo/open_model_zoo.py

```diff
@@ -0,0 +1,193 @@
+
+        model_dir = download_dir / model.subdirectory
+
+        flags = ['--name=' + model_name,
+                 '--output_dir=' + str(download_dir),
+                 '--cache_dir=' + str(cache_dir),
+                 '--precisions=' + str(precision)]
+
+        omz_downloader.download(flags)
+
+        description = '{}\n\n    License: {}'.format(model.description, model.license_url)
+        task_type = model.task_type
+
+        if precision not in model.precisions:
+            raise ValueError('Incorrect precision value!\n    Current precision value is: \''
+                                        + str(precision) + '\'.\n    Allowed precision values for model are: '
+                                        + str(model.precisions))
+
+        prefix = Path(model_dir) / precision / model_name
+
+        model_path = str(prefix) + '.xml'
+        bin_path = str(prefix) + '.bin'
+        name = model.name
+        subdirectory = model.subdirectory
+
+        if not os.path.exists(model_path) or not os.path.exists(bin_path):
+            omz_converter.converter(['--name=' + model_name, '--precisions=' + precision,
+                        '--download_dir=' + str(download_dir)])
+        return cls(name, model_path, description, task_type, subdirectory)
+
+    @classmethod
+    def from_pretrained(cls, model_path, *, task_type=None):
+        '''
+        Loads model from existing .xml, .onnx files.
+        Parameters
+        ----------
+            model_path
+                Path to .xml or .onnx file.
+            task_type
+                Type of task that the model performs.
+        '''
+        model_path = Path(model_path).resolve()
+
+        if not model_path.exists():
+            raise ValueError('Path {} to model file does not exist.'.format(model_path))
+
+        if model_path.suffix not in ['.xml', '.onnx']:
+            raise ValueError('Unsupported model format {}. Only .xml or .onnx supported.'.format(model_path.suffix))
+
+        description = 'Pretrained model {}'.format(model_path.name)
+        name = model_path.name
+        task_type = task_type
+        subdirectory = model_path.parent
+        model_path = str(model_path)
+
+        return cls(name, model_path, description, task_type, subdirectory)
+
+    def _load(self, ie, device):
+        self.net = ie.read_network(self.model_path)
+        self.exec_net = ie.load_network(self.net, device)
+
+    def __call__(self, inputs, ie, device='CPU'):
+        if self.exec_net is None:
+            self._load(ie, device)
+
+        input_names = next(iter(self.net.input_info))
+        for input_name, value in inputs.items():
+            if input_name not in input_names:
+                raise ValueError('Unknown input name {}'.format(input_name))
+
+            input_shape = self.net.input_info[input_name].input_data.shape
+            if input_shape != list(value.shape):
+                value = cv2.resize(value, input_shape)
```

Comment:
not sure that it will work for all cases, what about non-vision models?

Reply:
and BTW, even when input is image, usually input tensor is 4D (NCHW or NHWC), while opencv expects that image is 3D HWC np array

so I think we should avoid to use any preprocessing on our side and expect that user already prepare input tensor, or handle it more generally

Reply:
Done